### ISSUE TITLE
assertion 'p.destination_locality() == hpx::get_locality()' failed

### ISSUE THREAD

**Issue by dmarce1** [STEllAR-GROUP/hpx#1944]
I got an assertion failure while running octo-tiger

{what}: assertion 'p.destination_locality() == hpx::get_locality()' failed: HPX(assertion_failure)

Full program output is here:
[fail.txt](https://github.com/STEllAR-GROUP/hpx/files/79405/fail.txt)

(Note this is running with agas caching turned -off-. I found the performance degradation of https://github.com/STEllAR-GROUP/hpx/issues/1753 doesn't happen with caching turned off)

**Comment by hkaiser** [comment]
The relevant output is here:

```
{stack-trace}: 3 frames:
0x2aefc29ade48  : hpx::termination_handler(int) + 0x158 in /home/dmarce1/debug/lib/libhpxd.so.0
0x307240f710    : ??? + 0x307240f710 in /lib64/libpthread.so.0
{what}: Segmentation fault
{config}:
  HPX_HAVE_NATIVE_TLS=ON
  HPX_HAVE_STACKTRACES=ON
  HPX_HAVE_COMPRESSION_BZIP2=OFF
  HPX_HAVE_COMPRESSION_SNAPPY=OFF
  HPX_HAVE_COMPRESSION_ZLIB=OFF
  HPX_HAVE_PARCEL_COALESCING=ON
  HPX_HAVE_PARCELPORT_TCP=ON
  HPX_HAVE_PARCELPORT_MPI=ON (MPICH V3.1.2, MPI V3.0)
  HPX_HAVE_PARCELPORT_IPC=OFF
  HPX_HAVE_PARCELPORT_IBVERBS=OFF
  HPX_HAVE_VERIFY_LOCKS=ON
  HPX_HAVE_HWLOC=ON
  HPX_HAVE_ITTNOTIFY=OFF
  HPX_HAVE_RUN_MAIN_EVERYWHERE=OFF
  HPX_PARCEL_MAX_CONNECTIONS=512
  HPX_PARCEL_MAX_CONNECTIONS_PER_LOCALITY=4
  HPX_INITIAL_AGAS_LOCAL_CACHE_SIZE=256
  HPX_AGAS_LOCAL_CACHE_SIZE_PER_THREAD=32
  HPX_HAVE_MALLOC=custom
  HPX_PREFIX (configured)=/work/dmarce1/builds/hpx_build
  HPX_PREFIX=/home/dmarce1/debug
{version}: V0.9.12-trunk (AGAS: V3.0), Git: unknown
{boost}: V1.59.0
{build-type}: debug
{date}: Jan  4 2016 11:22:19
{platform}: linux
{compiler}: Intel C++ C++0x mode version 1500
{stdlib}: GNU libstdc++ version 20140422

{stack-trace}: 4 frames:
0x2b163361dd8f  : hpx::detail::backtrace(unsigned long) + 0x7f in /home/dmarce1/debug/lib/libhpxd.so.0
0x2b163361f3c8  : boost::exception_ptr hpx::detail::get_exception<hpx::exception>(hpx::exception const&, std::string const&, std::string const&, long, std::string const&) + 0x238 in /home/dmarce1/debug/lib/libhpxd.so.0
0x2b163361cf12  : hpx::detail::assertion_failed_msg(char const*, char const*, char const*, char const*, long) + 0x432 in /home/dmarce1/debug/lib/libhpxd.so.0
0x2b163361d9e2  : hpx::detail::assertion_failed(char const*, char const*, char const*, long) + 0x12 in /home/dmarce1/debug/lib/libhpxd.so.0
{env}: 141 entries:
  ACLOCAL_PATH=/usr/local/packages/cmake/2.8.12/INTEL-14.0.2/share/aclocal
  BASH_FUNC_module()=() {  eval `/usr/local/packages/Modules/$MODULE_VERSION/bin/modulecmd bash $*`
}
  CPATH=/home/dmarce1/debug/include:/usr/local/packages/python/2.7.7-anaconda/include:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//include:/usr/local/compilers/gcc/4.9.0/include:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/include:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mkl/include:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/tbb/include:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/include:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/tbb/include
  CUDA_NIC_INTEROP=1
  CVS_RSH=ssh
  ENVIRONMENT=BATCH
  GDBSERVER_MIC=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/target/mic/bin/gdbserver
  GDB_CROSS=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/bin/gdb-mic
  GDB_HOST=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/bin/gdb-ia-mic
  GFORTRAN_UNBUFFERED_PRECONNECTED=y
  G_BROKEN_FILENAMES=1
  HISTCONTROL=ignoredups
  HISTSIZE=1000
  HOME=/home/dmarce1
  HPX_NODEFILE=node.list
  INFOPATH=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64/share/info/:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/share/info/
  INTEL_LICENSE_FILE=/usr/local/compilers/Intel
  INTEL_PYTHONHOME=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/python/intel64/
  IPPROOT=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp
  I_MPI_CMD=mpirun -np 64 --machinefile node.list ./octotiger restart.chk -t20 -Ihpx.agas.use_caching=0 
  I_MPI_EXT_MPI_CALLS_29317=0x2b16385ae110
  I_MPI_HYDRA_RMK=pbs
  I_MPI_INFO_BRAND=Intel(R) Xeon(R) 
  I_MPI_INFO_CACHE1=0,1,2,3,4,8,9,10,11,12,16,17,18,19,20,24,25,26,27,28
  I_MPI_INFO_CACHE2=0,1,2,3,4,8,9,10,11,12,16,17,18,19,20,24,25,26,27,28
  I_MPI_INFO_CACHE3=0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1
  I_MPI_INFO_CACHES=3
  I_MPI_INFO_CACHE_SHARE=2,2,32
  I_MPI_INFO_CACHE_SIZE=32768,262144,26214400
  I_MPI_INFO_CORE=0,1,2,3,4,8,9,10,11,12,0,1,2,3,4,8,9,10,11,12
  I_MPI_INFO_C_NAME=Unknown
  I_MPI_INFO_DESC=1342177285
  I_MPI_INFO_FLGB=641
  I_MPI_INFO_FLGC=2143216639
  I_MPI_INFO_FLGD=-1075053569
  I_MPI_INFO_LCPU=20
  I_MPI_INFO_MODE=263
  I_MPI_INFO_NC=20
  I_MPI_INFO_NP=2
  I_MPI_INFO_NUMA_NODE_DIST=10,21,21,10
  I_MPI_INFO_NUMA_NODE_MAP=mlx4_0:1,scif0:-1,mic0:0,mic1:1
  I_MPI_INFO_NUMA_NODE_NUM=2
  I_MPI_INFO_PACK=0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1
  I_MPI_INFO_SERIAL=E5-2680 v2 
  I_MPI_INFO_SIGN=198372
  I_MPI_INFO_STATE=0
  I_MPI_INFO_THREAD=0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
  I_MPI_INFO_VEND=1
  I_MPI_MPIRUN=mpirun
  I_MPI_NUMBER_OF_MPI_PROCESSES_PER_NODE=1
  I_MPI_PERHOST=allcores
  I_MPI_PIN_INFO=x0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
  I_MPI_PIN_MAPPING=1:6 0
  I_MPI_PM=hydra
  I_MPI_RANK_CMD=./octotiger restart.chk -t20 -Ihpx.agas.use_caching=0 
  I_MPI_ROOT=/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048
  I_MPI_THREAD_LEVEL=0
  LANG=en_US.UTF-8
  LANGUAGE_TERRITORY=en_US
  LD_INCLUDE_PATH=/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/include
  LD_LIBRARY_PATH=/home/dmarce1/debug/lib:/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/intel64/lib:/usr/local/packages/python/2.7.7-anaconda/lib:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//lib:/usr/local/compilers/gcc/4.9.0/lib/gcc/x86_64-unknown-linux-gnu/4.9.0:/usr/local/compilers/gcc/4.9.0/lib64:/usr/local/compilers/gcc/4.9.0/lib:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/intel64:/opt/intel/mic/coi/host-linux-release/lib:/opt/intel/mic/myo/lib:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mpirt/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/../compiler/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/tools/intel64/perfsys:/opt/intel/mic/coi/host-linux-release/lib:/opt/intel/mic/myo/lib:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mkl/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/tbb/lib/intel64/gcc4.4:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/libipt/intel64/lib:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/compiler/lib/intel64:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/intel64/lib:/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/mic/lib
  LESSOPEN=|/usr/bin/lesspipe.sh %s
  LHPC_DEVELCMAKE=/usr/local/packages/cmake/2.8.12/INTEL-14.0.2/easybuild/cmake-2.8.12-INTEL-14.0.2-easybuild-devel
  LHPC_DEVELGCC=/usr/local/compilers/gcc/4.9.0/easybuild/gcc-4.9.0-easybuild-devel
  LHPC_DEVELINTEL=/usr/local/packages/INTEL/14.0.2/easybuild/INTEL-14.0.2-easybuild-devel
  LHPC_ROOTCMAKE=/usr/local/packages/cmake/2.8.12/INTEL-14.0.2
  LHPC_ROOTGCC=/usr/local/compilers/gcc/4.9.0
  LHPC_ROOTINTEL=/usr/local/packages/INTEL/14.0.2
  LHPC_ROOTPYTHON=/usr/local/packages/python/2.7.7-anaconda
  LHPC_ROOT_HWLOC=/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2/
  LHPC_VERSIONCMAKE=2.8.12
  LHPC_VERSIONGCC=4.9.0
  LHPC_VERSIONINTEL=14.0.2
  LHPC_VERSIONPYTHON=2.7.7
  LHPC_VERSION_HWLOC=1.10.0
  LIBRARY_PATH=/home/dmarce1/debug/lib:/usr/local/packages/python/2.7.7-anaconda/lib:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//lib:/usr/local/compilers/gcc/4.9.0/lib64:/usr/local/compilers/gcc/4.9.0/lib:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/../compiler/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mkl/lib/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/tbb/lib/intel64/gcc4.4:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/compiler/lib/intel64:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/lib/intel64
  LM_LICENSE_FILE=/usr/local/compilers/pgi/license.dat
  LOADEDMODULES=gcc/4.9.0:intel/14.0.2:INTEL/14.0.2:cmake/2.8.12/INTEL-14.0.2:hwloc/1.10.0/INTEL-14.0.2:python/2.7.7-anaconda
  LOGNAME=dmarce1
  MAIL=/var/spool/mail/dmarce1
  MANPATH=/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/man:/usr/local/packages/python/2.7.7-anaconda/share/man:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//share/man:/usr/local/packages/cmake/2.8.12/INTEL-14.0.2/man:/usr/local/compilers/gcc/4.9.0/share/man:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/man/en_US:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/man/en_US:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64/share/man/:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/share/man/::/usr/local/compilers/pgi/linux86-64/man:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/man/en_US:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/man/en_US
  MIC_LD_LIBRARY_PATH=/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/compiler/lib/mic:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/lib/mic
  MIC_LIBRARY_PATH=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/mic:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mpirt/lib/mic:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/tbb/lib/mic
  MKLROOT=/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl
  MOABHOMEDIR=/usr/moab
  MODULEPATH=/usr/local/packages/Modules/modulefiles/apps:/usr/local/packages/Modules/modulefiles/admin:/usr/local/packages/Modules/modulefiles/xsede
  MODULESHOME=/usr/local/packages/Modules/3.2.10
  MODULE_VERSION=3.2.10
  MODULE_VERSION_STACK=3.2.10
  MPIR_CVAR_CH3_INTERFACE_HOSTNAME=smic252
  MPIR_CVAR_NEMESIS_ENABLE_CKPOINT=1
  MPM_LAUNCHER=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/mpm/bin/start_mpm.sh
  NLSPATH=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/compiler/lib/intel64/locale/%l_%t/%N:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/ipp/lib/intel64/locale/%l_%t/%N:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/mkl/lib/intel64/locale/%l_%t/%N:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/share/locale/%l_%t/%N:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64/share/locale/%l_%t/%N:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/compiler/lib/intel64/locale/en_US:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/mkl/lib/intel64/locale/en_US
  NPROCS=64
  PATH=/home/dmarce1/debug/bin:/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/intel64/bin:/usr/local/packages/python/2.7.7-anaconda/bin:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//bin:/usr/local/packages/cmake/2.8.12/INTEL-14.0.2/bin:/usr/local/compilers/gcc/4.9.0/bin:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/bin/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/bin:/usr/lib64/qt-3.3/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/compilers/pgi/linux86-64/6.1/bin:/opt/ibutils/bin:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/bin/intel64:/home/dmarce1/bin
  PBS_ENVIRONMENT=PBS_BATCH
  PBS_GPUFILE=/var/spool/torque/aux//80123.smic3gpu
  PBS_JOBCOOKIE=F1B1BAE8FFB42FD77F91967C1A723ABE
  PBS_JOBID=80123.smic3
  PBS_JOBNAME=job
  PBS_MOMPORT=15003
  PBS_NODEFILE=node.list
  PBS_NODENUM=0
  PBS_NP=1280
  PBS_NUM_NODES=64
  PBS_NUM_PPN=20
  PBS_O_HOME=/home/dmarce1
  PBS_O_HOST=smic1.hpc.lsu.edu
  PBS_O_LANG=en_US.UTF-8
  PBS_O_LOGNAME=dmarce1
  PBS_O_MAIL=/var/spool/mail/dmarce1
  PBS_O_PATH=/home/dmarce1/debug/bin:/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/intel64/bin:/usr/local/packages/python/2.7.7-anaconda/bin:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//bin:/usr/local/packages/cmake/2.8.12/INTEL-14.0.2/bin:/usr/local/compilers/gcc/4.9.0/bin:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/bin/intel64:/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/debugger/gdb/intel64_mic/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/compilers/pgi/linux86-64/6.1/bin:/opt/ibutils/bin:/opt/dell/srvadmin/bin:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/bin/intel64:/home/dmarce1/bin
  PBS_O_QUEUE=checkpt
  PBS_O_SHELL=/bin/bash
  PBS_O_WORKDIR=/work/dmarce1/octotiger
  PBS_QUEUE=checkpt
  PBS_SERVER=smic3
  PBS_TASKNUM=1
  PBS_VERSION=TORQUE-3.0.6
  PBS_VNODENUM=0
  PBS_WALLTIME=86400
  PGI=/usr/local/compilers/pgi
  PKG_CONFIG_PATH=/usr/local/packages/python/2.7.7-anaconda/lib/pkgconfig:/usr/local/packages/hwloc/1.10.0/INTEL-14.0.2//lib/pkgconfig:/home/dmarce1/debug/lib/pkgconfig
  PMI_FD=7
  PMI_RANK=6
  PMI_SIZE=64
  PWD=/work/dmarce1/octotiger
  PYTHONPATH=/usr/local/packages/python/2.7.7-anaconda/lib/python2.7/site-packages
  QTDIR=/usr/lib64/qt-3.3
  QTINC=/usr/lib64/qt-3.3/include
  QTLIB=/usr/lib64/qt-3.3/lib
  SHELL=/bin/bash
  SHLVL=3
  SSH_ASKPASS=/usr/libexec/openssh/gnome-ssh-askpass
  SSH_CLIENT=204.90.41.103 41378 22
  SSH_CONNECTION=204.90.41.103 41378 204.90.41.252 22
  TBBROOT=/usr/local/compilers/Intel/parallel_studio-2015u3/composer_xe_2015.3.187/tbb
  USER=dmarce1
  _=/usr/local/compilers/Intel/parallel_studio-2015u3//impi/5.0.3.048/intel64/bin/mpiexec.hydra
  _LMFILES_=/usr/local/packages/Modules/modulefiles/apps/gcc/4.9.0:/usr/local/packages/Modules/modulefiles/apps/intel/14.0.2:/usr/local/packages/Modules/modulefiles/admin/INTEL/14.0.2:/usr/local/packages/Modules/modulefiles/apps/cmake/2.8.12/INTEL-14.0.2:/usr/local/packages/Modules/modulefiles/apps/hwloc/1.10.0/INTEL-14.0.2:/usr/local/packages/Modules/modulefiles/apps/python/2.7.7-anaconda
  _MODULESBEGINENV_=/home/dmarce1/.modulesbeginenv
{locality-id}: 6
{hostname}: [ (mpi:6) (tcp:204.90.41.252:7916) ]
{process-id}: 29317
{function}: void hpx::parcelset::decode_message(Parcelport &, Buffer, unsigned long, unsigned long) [with Parcelport = hpx::parcelset::policies::mpi::parcelport, Buffer = hpx::parcelset::parcel_buffer<std::vector<char, std::allocator<char>>, std::vector<char, std::allocator<char>>>]
{file}: /work/dmarce1/builds/hpx/hpx/runtime/parcelset/decode_parcels.hpp
{line}: 124
{os-thread}: worker-thread#5
{state}: state_running
{auxinfo}: 
{config}:
  HPX_HAVE_NATIVE_TLS=ON
  HPX_HAVE_STACKTRACES=ON
  HPX_HAVE_COMPRESSION_BZIP2=OFF
  HPX_HAVE_COMPRESSION_SNAPPY=OFF
  HPX_HAVE_COMPRESSION_ZLIB=OFF
  HPX_HAVE_PARCEL_COALESCING=ON
  HPX_HAVE_PARCELPORT_TCP=ON
  HPX_HAVE_PARCELPORT_MPI=ON (MPICH V3.1.2, MPI V3.0)
  HPX_HAVE_PARCELPORT_IPC=OFF
  HPX_HAVE_PARCELPORT_IBVERBS=OFF
  HPX_HAVE_VERIFY_LOCKS=ON
  HPX_HAVE_HWLOC=ON
  HPX_HAVE_ITTNOTIFY=OFF
  HPX_HAVE_RUN_MAIN_EVERYWHERE=OFF
  HPX_PARCEL_MAX_CONNECTIONS=512
  HPX_PARCEL_MAX_CONNECTIONS_PER_LOCALITY=4
  HPX_INITIAL_AGAS_LOCAL_CACHE_SIZE=256
  HPX_AGAS_LOCAL_CACHE_SIZE_PER_THREAD=32
  HPX_HAVE_MALLOC=custom
  HPX_PREFIX (configured)=/work/dmarce1/builds/hpx_build
  HPX_PREFIX=/home/dmarce1/debug
{version}: V0.9.12-trunk (AGAS: V3.0), Git: unknown
{boost}: V1.59.0
{build-type}: debug
{date}: Jan  4 2016 11:22:19
{platform}: linux
{compiler}: Intel C++ C++0x mode version 1500
{stdlib}: GNU libstdc++ version 20140422
{what}: assertion 'p.destination_locality() == hpx::get_locality()' failed: HPX(assertion_failure)

```

**Comment by hkaiser** [comment]
If this is what it looks like, then a parcel was delivered to the wrong locality. 

Is there anything unusual in the way you run your application? Any configuration settings? Command line arguments?

**Comment by dmarce1** [comment]
Other than turning off the agas caching there isn't anything unusual about this run. It is in 'debug' mode.

**Comment by sithhell** [comment]
On 01/06/2016 07:12 PM, Dominic Marcello wrote:
> Other than turning off the agas caching there isn't anything unusual
> about this run. It is in 'debug' mode.

I am not seeing any increased stack sizes, could this be a stack 
overflow problem again?

**Comment by dmarce1** [comment]
I declared every component action using HPX_ACTION_USES_HUGE_STACK,

**Comment by hkaiser** [comment]
Do you realize that 'huge stack' means 32MBytes? That's 32 times more than a normal Linux thread has available. I wouldn't be surprised if this alone causes the slowdowns you're seeing.

**Comment by dmarce1** [comment]
I re-ran with "large" stacks (2 MB?) and got the similar results.

**Comment by hkaiser** [comment]
This problem was seen in the wild in other contexts as well, but again related to disabled AGAS caching (see #1966).

**Comment by hkaiser** [comment]
This was fixed by merging #1966
